src/modules/audit/components/auditHistoryColumnDefs.tsx

```diff
@@ -0,0 +1,161 @@
+import { Link, Stack, Typography } from '@mui/material';
+import { compact, filter } from 'lodash-es';
+import {
+  ContextualCollapsibleList,
+  ContextualListExpansionButton,
+} from '@/components/elements/CollapsibleListContext';
+import { ColumnDef } from '@/components/elements/table/types';
+import AuditObjectChangesSummary, {
+  ObjectChangesType,
+} from '@/modules/audit/components/AuditObjectChangesSummary';
+import { hasMeaningfulValue } from '@/modules/form/util/formUtil';
+import RelativeDateTableCellContents from '@/modules/hmis/components/RelativeDateTableCellContents';
+import { auditActionForDisplay } from '@/modules/hmis/hmisUtil';
+import { Routes } from '@/routes/routes';
+import { AuditEventType } from '@/types/gqlTypes';
+import { generateSafePath } from '@/utils/pathEncoding';
+
+export type AuditHistoryNode = NonNullable<{
+  id: string;
+  createdAt: string;
+  event: AuditEventType;
+  objectChanges?: any | null;
+  recordName: string;
+  graphqlType: string;
+  recordId: string;
+  user?: {
+    id: string;
+    name: string;
+  } | null;
+  trueUser?: {
+    id: string;
+    name: string;
+  } | null;
+  clientId?: string;
+  clientName?: string;
+  projectId?: string;
+  projectName?: string;
+  enrollmentId?: string;
+}>;
+
+export const auditHistoryColumns: ColumnDef<AuditHistoryNode>[] = [
```

Comment:
Nice, thanks for pulling this out into its own file. 
I think the most convenient way I have found for managing the columns definitions so far is to specify shared columns [in an object](https://github.com/greenriver/hmis-frontend/blob/ea9fbca1ec3893fbed44cd8eee5c3f4f644ccb07/src/modules/projects/components/tables/ProjectClientEnrollmentsTable.tsx#L56).

When done like that, there are 2 benefits:
1. different tables can easily pick and choose which columns they want and in which order. they can also insert their own "custom " columns in whichever location they want. [like this!](https://github.com/greenriver/hmis-frontend/blob/ea9fbca1ec3893fbed44cd8eee5c3f4f644ccb07/src/modules/bedNights/components/ProjectEnrollmentsTableForBedNights.tsx#L58-L82)
2. we can use a union type to support different fragments when there are slight variations

Your union type would be: 
```
export type AnyAuditEvent =
  | ClientAuditEventFieldsFragment
  | EnrollmentAuditEventFieldsFragment
  | UserAuditEventFieldsFragment;
```
I think that should make typescript happy. The Client and Project column definitions are specific to the User Audit page, at least for now.

Reply:
Thanks for this suggestion. I think it's much better